### ISSUE TITLE
Upgrade to 1.15.0 causes limit error when DD_PROFILING_ENABLED

### ISSUE THREAD

**Issue by KylerWittingThredUP** [DataDog/dd-trace-java#5322]
Downgrading back to `14.0.0` resolves the issue while every version in release 15 this issue comes up. This is a spring boot application. Tell me if you need any more information. 

## Error
```
java: /go/src/github.com/SL95iHeW/0/DataDog/apm-reliability/async-profiler-build/java-profiler/ddprof-lib/src/main/cpp/buffers.h:66: void Buffer::put(const char*, u32): Assertion `_offset + len < limit()' failed.
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f484bd90898, pid=1, tid=47
#
# JRE version: OpenJDK Runtime Environment Temurin-17.0.7+7 (17.0.7+7) (build 17.0.7+7)
# Java VM: OpenJDK 64-Bit Server VM Temurin-17.0.7+7 (17.0.7+7, mixed mode, sharing, tiered, compressed oops, compressed class ptrs, g1 gc, linux-amd64)
# Problematic frame:
# C  [libc.so.6+0x28898]  abort+0x178
#
# Core dump will be written. Default location: /app/core.1
#
# JFR recording file will be written. Location: /app/hs_err_pid1.jfr
#
# An error report file with more information is saved as:
# /app/hs_err_pid1.log
#
# If you would like to submit a bug report, please visit:
#   https://github.com/adoptium/adoptium-support/issues
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
```

**Comment by KylerWittingThredUP** [comment]
@richardstartin 

So it looks like the `hs_err_pid1.log` isn't actually getting created :( I am unable to find any other logs really to help debug this on the pod

```
#!/bin/sh

java \
  -Djava.security.properties=security.properties \
  -javaagent:dd-java-agent.jar \
  $JAVA_OPTIONS \
  org.springframework.boot.loader.JarLauncher \
; ls /app/* 2>/dev/null \
; cat /app/hs_err_pid1.log 2>/dev/null \
```

```
/app/core.7
/app/dd-java-agent.jar
/app/security.properties

/app/BOOT-INF:
classes
classpath.idx
layers.idx
lib

/app/META-INF:
build-info.properties
MANIFEST.MF
maven
spring.components
spring-configuration-metadata.json

/app/org:
springframework
```

**Comment by richardstartin** [comment]
We believe this was caused by long thread names overflowing a buffer when recording profiles. This should be fixed in 1.16.0 next week.

**Comment by richardstartin** [comment]
1.16.0 has been released with a fix for this bug. Please reopen if it recurrs.
